client/server/lib/memcached/index.ts

```diff
@@ -0,0 +1,47 @@
+import config from '@automattic/calypso-config';
+import { Client } from 'memjs';
+
+// Memcached should not be used in the client or environments where the server is
+// not configured.
+const MEMCACHED_DISABLE = typeof document !== 'undefined' || ! config( 'server_side_memcached' );
+let MEMCACHED_CLIENT: Client | undefined;
+
+const ENV = config( 'env_id' ) as string;
+
+export function createMemcachedClient() {
+	if ( MEMCACHED_DISABLE || MEMCACHED_CLIENT ) {
+		return;
+	}
+	// todo: do we need to handle disconnects/reconnects?
```

Comment:
In theory, we should handle disconnects. I'd doubt it would happen within a single server page load but who knows? Maybe we can add some logging to check if a disconnect can ever occur.

Reply:
It looks like the library handles some of this:

https://github.com/memcachier/memjs/blob/647a2e1c26b9a85b2f134ef332aeb669eb5cc023/lib/memjs/server.js#L152-L174

I'm not really sure what we need to do on top of that.